src/Mod/TechDraw/Gui/CommandAlign.cpp

```diff
@@ -0,0 +1,158 @@
+    sToolTipText    = sMenuText;
+    sWhatsThis      = "TechDraw_AlignGroup";
+    sStatusTip      = sToolTipText;
+}
+
+void CmdTechDrawAlignVertexesVertically::activated(int iMsg)
+{
+    TechDraw::DrawViewPart* dvp = static_cast<TechDraw::DrawViewPart*>(
+        getSelection().getObjectsOfType(TechDraw::DrawViewPart::getClassTypeId())[0]
+    );
+
+    Gui::Document* guiDoc = Gui::Application::Instance->getDocument(dvp->getDocument());
+    if (!guiDoc) {
+        return;
+    }
+
+    Gui::ViewProvider* gvp = guiDoc->getViewProvider(dvp);
+    auto vpdvp = static_cast<TechDrawGui::ViewProviderViewPart*>(gvp);
+    if (!vpdvp) {
+        return;
+    }
+    TechDrawGui::QGIView* view = vpdvp->getQView();
+
+    std::vector<std::string> vertexNames = getSelection().getSelectionEx()[0].getSubNames();
+    std::vector<int> vertexIndexes = DrawUtil::getIndexFromName(vertexNames);
+
+    std::vector<QGIVertex*> vertexes = view->getObjects<QGIVertex*>(vertexIndexes);
+    if(vertexes.size() != 2) {
+        Base::Console().Error(QObject::tr("You must select 2 vertexes\n").toStdString().c_str());
+        return;
+    }
+    QGIVertex* v1 = vertexes.front();
+    QGIVertex* v2 = vertexes.back();
+    DrawGuiUtil::rotateToAlignVertically(v1, v2);
+}
+
+bool CmdTechDrawAlignVertexesVertically::isActive(void)
+{
+    bool havePage = DrawGuiUtil::needPage(this);
+    bool haveView = DrawGuiUtil::needView(this, false);
+    return (havePage && haveView);
+}
+
+
+//===========================================================================
+// TechDraw_AlignVertexesHorizontally
+//===========================================================================
+
+DEF_STD_CMD_A(CmdTechDrawAlignVertexesHorizontally)
+
+CmdTechDrawAlignVertexesHorizontally::CmdTechDrawAlignVertexesHorizontally()
+  : Command("TechDraw_AlignVertexesHorizontally")
+{
+    sAppModule      = "TechDraw";
+    sGroup          = QT_TR_NOOP("TechDraw");
+    sMenuText       = QT_TR_NOOP("Align vertexes horizontally by view rotation");
+    sToolTipText    = sMenuText;
+    sWhatsThis      = "TechDraw_AlignGroup";
+    sStatusTip      = sToolTipText;
+}
+
+void CmdTechDrawAlignVertexesHorizontally::activated(int iMsg)
+{
+    TechDraw::DrawViewPart* dvp = static_cast<TechDraw::DrawViewPart*>(
+        getSelection().getObjectsOfType(TechDraw::DrawViewPart::getClassTypeId())[0]
+    );
+
+    Gui::Document* guiDoc = Gui::Application::Instance->getDocument(dvp->getDocument());
+    if (!guiDoc) {
+        return;
+    }
+
+    Gui::ViewProvider* gvp = guiDoc->getViewProvider(dvp);
+    auto vpdvp = static_cast<TechDrawGui::ViewProviderViewPart*>(gvp);
+    if (!vpdvp) {
+        return;
+    }
+    TechDrawGui::QGIView* view = vpdvp->getQView();
+
+    std::vector<std::string> vertexNames = getSelection().getSelectionEx()[0].getSubNames();
+    std::vector<int> vertexIndexes = DrawUtil::getIndexFromName(vertexNames);
+
```

Comment:
Didn't the feature request ask for alignment by an edge?  This only does 2 vertexes.  Need something like:

    std::string gType = TechDraw::DrawUtil::getGeomTypeFromName(subnames.at(0));
    if (gType == "Edge") {.
        auto edgeIndex = TechDraw::DrawUtil::getIndexFromName(Name);
        TechDraw::BaseGeomPtr geom = dvp->getGeomByIndex(edgeIndex);
        if (geom->getGeomType() != TechDraw::GENERIC) {
            QMessageBox::warning(Gui::getMainWindow(), QObject::tr("Incorrect Selection"),
                        QObject::tr("Please select a straight edge"));
            return;
        }
        point0 = geom->getStartPoint();
        point1 = geom->getEndPoint();
    } else if (gType == "Vertex") {
        ...
    }

Reply:
It indeed did! However, I think that it will give more flexibility with points. There might not be an edge for the desired alignment, but there will always be points for a desired edge.

We can also do, that either two points or one edge are required?